include/boost/beast/experimental/core/url_parser.hpp

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright (c) 2018 jackarain (jack dot wgm at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_BEAST_EXPERIMENTAL_CORE_URL_PARSER_HPP
+#define BOOST_BEAST_EXPERIMENTAL_CORE_URL_PARSER_HPP
+
+#include <boost/beast/core/string.hpp>
+#include <cctype> // for std::isalpha etc...
+
+namespace boost {
+namespace beast {
+
+    //--------------------------------------------------------------------------
+    // scheme:[//[user[:password]@]host[:port]][/path][?query][#fragment]
+
+    using boost::beast::string_view;
+
+    struct url_parser
+    {
+        string_view scheme_;
```

Comment:
MInor point, public members don't end with an underscore, e.g. `scheme` instead of `scheme_`.

Reply:
Noted, will change.